pinot-controller/src/main/java/org/apache/pinot/controller/validation/StorageQuotaChecker.java

```diff
@@ -205,4 +213,48 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(TableConfig tableConfig,
       return failure(message);
     }
   }
+
+  public boolean isTableStorageQuotaExceeded(TableConfig tableConfig) {
```

Comment:
Should we also check if quota check is enabled?
Consider extracting the common part for `isSegmentStorageWithinQuota()` and this method

Reply:
Agree, will do.